include/meta/meta.hpp

```diff
@@ -111,6 +110,11 @@ namespace meta
     template <std::size_t N>
     using size_t = std::integral_constant<std::size_t, N>;
 
+    /// A metafunction that computes the size of the type \p T.
+    /// \par Complexity
+    /// \f$ O(1) \f$.
+    template <class T> using size_of = meta::size_t<sizeof(T)>;
```

Comment:
The idiom I use for names for things that are already keywords has been to add a trailing underscore, so `sizeof_`, but `size_of` seems OK, too.